Sources/DeepLearning/Optimizer.swift

```diff
@@ -23,14 +23,22 @@ import TensorFlow
 public protocol Optimizer {
     /// The type of the model whose parameters are optimized.
     associatedtype Model: Layer
+    /// The scalar parameter type.
+    associatedtype Scalar: FloatingPoint
     /// The learning rate.
-    var learningRate: Float { get set }
+    var learningRate: Scalar { get set }
     /// Updates the specified differentiable variables along the specified
     /// direction.
     mutating func update(_ variables: inout Model.AllDifferentiableVariables,
                          along direction: Model.CotangentVector)
 }
 
+fileprivate extension Tensor where Scalar: Numeric {
+    mutating func resetToZero() {
```

Comment:
Nit: I'd say `formZero` is a more Swifty name here. `reset` is a bit unprecedented.

https://swift.org/documentation/api-design-guidelines/#naming